tasks/ModelTask.cpp

```diff
@@ -30,101 +29,102 @@ void ModelTask::setGazeboModel(WorldPtr _world,  ModelPtr _model)
+            }
+        }
 	}
 }
-//======================================================================================
+
 void ModelTask::updateHook()
 {
     updateJoints();
     updateLinks();
 }
-//======================================================================================
+
 void ModelTask::updateJoints()
 {
-	for(JointPort_V::iterator it = joint_port_list.begin();it != joint_port_list.end(); ++it)
-	{
-		double effort = 0;	
-		(*it).first->readNewest(effort);
-		if(effort != RTT::NoData)
-		{
-			// Define the limits for the joint effort
-			try{
-			    if((-1.0 <= effort) && (effort <= 1.0)){
-			        (*it).second->SetForce(0,effort);
-			    }else{
-			        throw;
-			    }
-			}catch(...){
-			    // Joint effort value out of range 
-			    gzmsg << "RockBridge: error - effort value out of range (-1 <= effort <= 1)." << std::endl;
-			}
-		}
-	}
+    _joints_cmd.readNewest( joints_in );
+    for(Joint_V::iterator it = gazebo_joints.begin(); it != gazebo_joints.end(); ++it )
+    {
+        // Apply effort to joint
+        double effort = joints_in.getElementByName( (*it)->GetName() ).effort;
+        if( effort != RTT::NoData )
+            (*it)->SetForce(0,effort);
+
+        // Read joint angle from gazebo link
+        double angle = (*it)->GetAngle(0).Radian();
+        joints_out.getElementByName( (*it)->GetName() ).Position(angle);
+    }
+    _joints_samples.write( joints_out );
 }
-//======================================================================================
+
 void ModelTask::updateLinks()
 {
-	// Update all links from gazebo model
-	for(LinkPort_V::iterator it = link_port_list.begin(); it != link_port_list.end(); ++it)
-	{
-		LinkPtr link = (*it).second;
-
-		// Read rock input port and apply a force to gazebo links
-		base::Vector3d force(0.0, 0.0, 0.0);
-		(*it).first->readNewest(force);				
-		if((force(0) != RTT::NoData)||(force(1) != RTT::NoData)||(force(2) != RTT::NoData)){
-			link->AddRelativeForce( math::Vector3(force(0),force(1),force(2)) );
-		}
-	}
+    base::samples::RigidBodyState rock_rbs;
+    for(LinkOutput::iterator it = link_list.begin(); it != link_list.end(); ++it)
+    {
+        math::Pose link_pose = (*it).second->GetWorldPose();
+        rock_rbs.position = base::Vector3d(link_pose.pos.x,link_pose.pos.y,link_pose.pos.z);
+        rock_rbs.orientation = base::Quaterniond(
+                link_pose.rot.w,link_pose.rot.x,link_pose.rot.y,link_pose.rot.z );
+        (*it).first->write( rock_rbs );
+    }
 }
-//======================================================================================
+
+bool ModelTask::configureHook()
+{
+    if( ! ModelTaskBase::configureHook() )
+        return false;
+
+    // The robot configuration YAML file must define the link that will be exported.
+    // Than, these links will be loaded in link_names;
+    link_names = _exported_links.get();
+    setLinkPorts();
```

Comment:
since you have to call setLinkPorts here, also call setJointPorts. (That's really for consistency reasons).